infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationSinkManager.java

```diff
@@ -222,6 +236,27 @@ private void initWriterAndBufferMgr() {
                 metadataManager.getReplicationMetadata(session).getLastLogEntryBatchProcessed(), this);
     }
 
+    private void insertQInRegistryTable() {
+        TableRegistry tableRegistry = runtime.getTableRegistry();
+        String replicatedQName = REPLICATED_QUEUE_NAME_PREFIX + session.getSourceClusterId();
+        if (!tableRegistry.getRegistryTable().containsKey(replicatedQName)) {
+            try {
+                TableOptions tableOptions = TableOptions.builder()
+                        .schemaOptions(CorfuOptions.SchemaOptions.newBuilder()
+                                .addStreamTag(REPLICATED_QUEUE_TAG + DEFAULT_ROUTING_QUEUE_CLIENT)
```

Comment:
we can get the client name from the session.. no need to hard code it to the default value